extensions/blocks/map/map.php

```diff
@@ -25,7 +25,98 @@
 function jetpack_map_block_load_assets( $attr, $content ) {
 	$api_key = Jetpack_Options::get_option( 'mapbox_api_key' );
 
+	if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
+		static $map_block_counter = [];
+
+		$id = get_the_ID();
+		if ( ! $map_block_counter[ $id ] ) {
+			$map_block_counter[ $id ] = 0;
+		}
+		$map_block_counter[ $id ]++;
+
+		$iframe_url = add_query_arg(
+			array(
+				'map-block-counter' => $map_block_counter[ get_the_ID() ],
```

Comment:
Can we use `$id` here since we define it above?

```suggestion
				'map-block-counter' => $map_block_counter[ $id ],
```

Reply:
In addition to this, do you think it would be necessary to escape this and the line below? that may be overkill?

```suggestion
				'map-block-counter' => absint( $map_block_counter[ $id ] ),
```

Reply:
Yup, good ideas: https://github.com/Automattic/jetpack/pull/13405/commits/1000d1708adbd5fe42ae0959cce456c641b71ed3